### ISSUE TITLE
Unused track models

### ISSUE THREAD

**Issue by supertuxkart-sourceforge-migration** [supertuxkart/stk-code#1081]
**Author: curaga**

There are a total of 4mb of unused b3d files in trunk. They are not mentioned in any xml file for the track.

```
alpha-planetarium/sun-2.b3d
farm/cow_LOD120.b3d
farm/horse_LOD100.b3d
farm/skull_LOD120.b3d
farm/well_LOD75.b3d
fortmagma/lavaspike.001.b3d
fortmagma/Plane.043.b3d
fortmagma/Plane.044.b3d
hacienda/grass.jpg_track.b3d
introcutscene/camera.b3d
introcutscene/catailVeg_LOD50.b3d
introcutscene/introcutscenes_track.b3d
introcutscene/Remote.b3d
introcutscene2/pineVeryLow.b3d
introcutscene2/ShipMouth.b3d
introcutscene2/terrain.b3d
lighthouse/grass_chunk.b3d
lighthouse/miniLampPost.b3d
lighthouse/oceanBed.b3d
lighthouse/waterTop.b3d
mansion/clounds.b3d
mansion/Plane.003.b3d
mansion/Plane.008.b3d
mansion/Plane.009.b3d
mansion/Plane.014.b3d
mansion/Plane.015.b3d
mansion/Plane.016.b3d
mansion/Plane.017.b3d
mansion/Plane.018.b3d
mansion/Plane.019.b3d
mansion/Plane.020.b3d
mansion/Plane.026.b3d
mansion/Plane.027.b3d
mansion/Plane.028.b3d
mansion/Plane.029.b3d
mansion/Plane.030.b3d
mansion/Plane.b3d
mines/Circle.036_Circle.001.b3d
mines/cogwheel5.b3d
mines/lamp2.b3d
mines/lantern_LODx.b3d
minigolf/Icosphere.b3d
minigolf/testblok.b3d
minigolf/windmill_arm2.b3d
overworld/fern_LOD30.b3d
overworld/littleTree.b3d
overworld/pinetreeDetailed.b3d
overworld/terrain1_splat.b3d
overworld/terrain2_splat.b3d
snowtuxpeak/Plane.006.b3d
snowtuxpeak/Plane.007.b3d
snowtuxpeak/Plane.008.b3d
snowtuxpeak/Plane.011.b3d
snowtuxpeak/Plane.012.b3d
snowtuxpeak/Plane.013.b3d
snowtuxpeak/Plane.014.b3d
snowtuxpeak/Plane.017.b3d
snowtuxpeak/Plane.018.b3d
snowtuxpeak/Plane.020.b3d
snowtuxpeak/Plane.023.b3d
snowtuxpeak/Plane.024.b3d
xr591/big.b3d
xr591/electri.b3d
xr591/generator.b3d
xr591/SomeModel.b3d
zengarden/directionLantern_LOD200.b3d
zengarden/pagoda_noLod.b3d
zengarden/Plane.007.b3d
zengarden/sakuraPine.b3d
zengarden/skylantern_group.b3d
```

Migrated-From: https://sourceforge.net/apps/trac/supertuxkart/ticket/1025

**Comment by hiker** [comment]
Track cleaning up should be done before the RC, We have also reports about unused textures in Cocoa (old skybox textures)?

**Comment by samuncle** [comment]
Hi, I agree some textures unused should be removed. For b3D the easiest solution would be when the track is exported to clean up all .bd3. So this case won't be possible anymore. Because otherwise the case were it happens is quite easy. You export a track with an object like hello.b3D, then for some reason you remove it and you end up with an unused object in the track/yourtrack directory.

**Comment by samuncle** [comment]
As I said in the previous post. The easiest way to do that is to add an automated cleanup in the exporter which remove unused .b3d. 

Sure I can remove manually unused objects but then when I change the name of one object in blender the problem will reaper.

**Comment by hiker** [comment]
More than just the b3d - textures as well.

**Comment by samuncle** [comment]
I still won't fix until the exporter clean up the files automatically. I already removed some old files in cocoa temple but that should be up to the exporter of doing this.

**Comment by hiker** [comment]
How is the exporter supposed to do that? Delete all files in the target directory? What if the user has files that need to be kept (e.g. I often store a .blend in the same directory I export to - admittedly based on older versions of the exporter where you had to re-select the export directory all the time).
Users might add textures in a directory to be used later, which would then be removed.
Either we have to follow a proper process (delete all files before exporting for anything that will be committed), or we need a very 'secured' way of removing files (maybe offering an option to clean the target directory, and if this is selected, still ask if this should really be done).

**Comment by auriamg** [comment]
It could simply delete all b3d files before exporting. I don't think we need to delete everything, that would indeed be dangerous - b3d files only is possible though

**Comment by auriamg** [comment]
The exporter can now delete b3d files before writing the new ones, so this should fix itsself as tracks are re-exported
